### PR TITLE
update create-plugin yarn command

### LINKED ISSUE #8581: yarn create-plugin --backend no longer works
## Expected Behavior

Docs for creating a backend plugin should be up-to-date.

## Current Behavior

`yarn create-plugin --backend` is no longer valid after https://github.com/backstage/backstage/pull/8039

## Possible Solution

Update docs https://backstage.io/docs/plugins/backend-plugin with the latest way to create a backend plugin i.e. `yarn backstage-cli create` and choose backend-plugin.

@rugvip Any other thoughts on this? 😁

### PR DESCRIPTION
**Pull request by pwamej**
## Hey, I just made a Pull Request!

Fixes #8581. Parameters values are left intact. This aligns with what is described in [docs/local-dev/cli-commands.md#create-plugin]

#### :heavy_check_mark: Checklist

- [X] Added or updated documentation
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/plugins/backend-plugin.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ This will think for a bit, and then say `Listening on :7007`. In a different
 terminal window, now run
 
 ```sh
-curl localhost:7007/carmen/health
+curl localhost:7007/carmen-backend/health
```
[the remark this answers is not in the document]
Comment (pwamej) [reply docs/plugins/backend-plugin.md:51]: Yes, after generating plugin using `backstage-cli create --scope backstage --select backend-plugin` with ID `spinnaker` the path `/spinnaker-backend/health` was working, and `/spinnaker/health` was not

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "lint:type-deps": "node scripts/check-type-dependencies.js",
     "docker-build": "yarn tsc && yarn workspace example-backend build --build-dependencies && yarn workspace example-backend build-image",
     "backstage-create": "backstage-cli create --scope backstage --no-private",
-    "create-plugin": "yarn backstage-create --select plugin",
+    "create-plugin": "backstage-cli create-plugin --scope backstage --no-private",
```
[the remark this answers is not in the document]
Comment (pwamej) [reply package.json]: ok, I've added `create-backend-plugin` script

File: docs/plugins/backend-plugin.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rugvip) [reply docs/plugins/backend-plugin.md:51]: Ah you're right! That's actually a bug in the templating. We're suffixing the `id` with `-backend`, but we really shouldn't be doing that. Do you want to contribute a fix for that instead? That is this documentation is correct, it's the behavior that's wrong 😅

File: package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "docker-build": "yarn tsc && yarn workspace example-backend build --build-dependencies && yarn workspace example-backend build-image",
     "backstage-create": "backstage-cli create --scope backstage --no-private",
     "create-plugin": "yarn backstage-create --select plugin",
+    "create-backend-plugin": "yarn backstage-create --select backend-plugin",
```
Comment (Rugvip) [inline package.json:28]: We don't really want to add this tbh, as we're migrating away from `create-plugin`.

The one migration we could get started though is to recommend the creation of a `backstage-create` script in all apps, as well as adding it to the `create-app` template. Then we could update the docs to point to `yarn backstage-cli --select backend-plugin` instead. If that's something you'd like to pick up

**Comment by pwamej** [comment]
Ok, I'll take a look at that.

File: package.json (hunk 3 above)
Comment (pwamej) [reply package.json:28]: Ok, I'll take a look at that.

File: docs/plugins/backend-plugin.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pwamej) [reply docs/plugins/backend-plugin.md:51]: Sure 🙂

File: package.json (hunk 3 above)
Comment (freben) [reply package.json:28]: Hey, just wanted to check in if this is still on your radar :)

**Pull request closed without merging**